### PR TITLE
Add AMO electrochemical nutrient recovery model

### PR DESCRIPTION
**Pull request by luohezhiming**
## Summary/Motivation:
Add electrochemical nutrient recovery model for AMO project 1617

## Changes proposed in this PR:
- add electroN_P model
- add corresponding tests
- fixed typo in mabr_zo.py

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/unit_models/zero_order/tests/test_electroN_P_zo.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+                    + sum(
+                        model.fs.unit.generation_rxn_comp[0, r, j]
+                        for r in model.fs.unit.reaction_set
+                    )
+                    - model.fs.unit.treated.flow_mass_comp[0, j]
+                    - model.fs.unit.byproduct.flow_mass_comp[0, j]
+                )
+            )
+
+    @pytest.mark.component
```
Comment (bknueven) [inline watertap/unit_models/zero_order/tests/test_electroN_P_zo.py]:
```suggestion
    @pytest.mark.requires_idaes_solver
    @pytest.mark.component
```

For the macOS tests. That said, I'm not really a fan of testing these string literals.

**Review by adam-a-a — DISMISSED** [review]
LGTM

File: watertap/data/techno_economic/tests/test_unit_parameter_files.py [hunk 2]
```diff
@@ -92,6 +92,8 @@ def test_unit_parameter_files(tech):
         "ozone_aop",
         "photothermal_membrane",
         "pump_electricity",
+        "CANDO_P",
+        "electroN_P",
```
Comment (bknueven) [inline watertap/data/techno_economic/tests/test_unit_parameter_files.py]: `"CANDO_P"` is already in the last.

We're also trying to keep list like these in alphabetical order

**Review by bknueven — COMMENTED** [review]
It would be nice if `electroN_P` were moved to the correct place in the `no_energy_electric_flow_vol_inlet` list. These things become impossible to maintain without some order.

As a style comment, it would be more consistent if `electroN_P` were replaced everywhere by `electro_np`.

**Comment by luohezhiming** [comment]
> It would be nice if `electroN_P` were moved to the correct place in the `no_energy_electric_flow_vol_inlet` list. These things become impossible to maintain without some order.
> 
> As a style comment, it would be more consistent if `electroN_P` were replaced everywhere by `electro_np`.

Thank you for the suggestion, I take the name `electroN_P` from the provided slide, they called it `eltroN-P`. I also think the `electro_np` might be a better option following the convention.

File: watertap/core/zero_order_costing.py [hunk 3]
```diff
@@ -804,6 +805,64 @@ def cost_dmbr(blk):
             blk.unit_model.electricity[t0], "electricity"
         )
 
+    def cost_electroN_P(blk):
```
Comment (bknueven) [inline watertap/core/zero_order_costing.py]:
```suggestion
    def cost_electro_np(blk):
```

File: watertap/core/zero_order_costing.py [hunk 4]
```diff
@@ -2135,6 +2194,7 @@ def _general_power_law_form(blk, A, B, sizing_term, factor=None):
         CoagulationFlocculationZO: cost_coag_and_floc,
         DeepWellInjectionZO: cost_deep_well_injection,
         DMBRZO: cost_dmbr,
+        ElectroNPZO: cost_electroN_P,
```
Comment (bknueven) [inline watertap/core/zero_order_costing.py]:
```suggestion
        ElectroNPZO: cost_electro_np,
```

File: watertap/data/techno_economic/tests/test_unit_parameter_files.py (hunk 2 above)
Comment (tarnold17) [reply watertap/data/techno_economic/tests/test_unit_parameter_files.py]: I will second that `CANDO_P` should only be included once in the list.

**Review by bknueven — DISMISSED** [review]

**Comment by adam-a-a** [comment]
@luohezhiming Another option for the name of the unit would be "electrochemical_nutrient_removal" which is even more in line with how we've been naming units. Not essential, but I'll let you make the choice on that

**Comment by luohezhiming** [comment]
> @luohezhiming Another option for the name of the unit would be "electrochemical_nutrient_removal" which is even more in line with how we've been naming units. Not essential, but I'll let you make the choice on that

I agree that providing the full-name of the model will be the most clear, will revise accordingly.

**Review by adam-a-a — APPROVED** [review]
LGTM

**Review by bknueven — APPROVED** [review]

**Pull request merged**